### ISSUE TITLE
[SCM-879] JazzChangeLogCommandTest#testListChangesetConsumerWithTimeOnly() fails with NPE on Java 10

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1104]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[SCM-879](https://issues.apache.org/jira/browse/SCM-879?redirect=false)** and commented

Surefire says:

```
Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.917 sec <<< FAILURE! - in org.apache.maven.scm.provider.jazz.command.changelog.JazzChangeLogCommandTest
testListChangesetConsumerWithTimeOnly(org.apache.maven.scm.provider.jazz.command.changelog.JazzChangeLogCommandTest)  Time elapsed: 0.678 sec  <<< ERROR!
java.lang.NullPointerException
        at org.apache.maven.scm.provider.jazz.command.changelog.JazzChangeLogCommandTest.testListChangesetConsumerWithTimeOnly(JazzChangeLogCommandTest.java:294)
```

The error is rooted in `org.apache.maven.scm.provider.jazz.command.changelog.JazzListChangesetConsumer.processModifiedLine(String)`. It does not parse the given time with `JAZZ_TIMESTAMP_PATTERN_TIME` **and** `Locale.ENGLISH`, but relies on the system locale which will fail for non-English parsing `AM/PM`.

---

**Affects:** 1.9.5

**Issue Links:**
- [SCM-880](https://issues.apache.org/jira/browse/SCM-880) Maven SCM Client fails to be assembled on Java 10
 (_**"is depended upon by"**_)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Branch pushed, verified to work on:

```
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T20:49:05+01:00)
Maven home: D:\Entwicklung\Programme\apache-maven-3.5.3\bin\..
Java version: 10, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk-10
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

```
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T20:49:05+01:00)
Maven home: D:\Entwicklung\Programme\apache-maven-3.5.3\bin\..
Java version: 1.7.0_171, vendor: Azul Systems, Inc.
Java home: C:\Program Files\Zulu\zulu-7\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

```
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T20:49:05+01:00)
Maven home: /opt/apache-maven-3.5.3
Java version: 1.8.0_151, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.151-1.b12.fc25.x86_64/jre
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "4.13.16-100.fc25.x86_64", arch: "amd64", family: "unix"
```

```
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T20:49:05+01:00)
Maven home: /opt/apache-maven-3.5.3
Java version: 10, vendor: Azul Systems, Inc.
Java home: /usr/lib/jvm/zulu-10
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "4.13.16-100.fc25.x86_64", arch: "amd64", family: "unix"
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [57c9e6caa823011f9c4cc9d0e72790cbc1f23e4a](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=57c9e6caa823011f9c4cc9d0e72790cbc1f23e4a).
